pkg/clusteragent/autoscaling/workload/telemetry.go

```diff
@@ -158,15 +456,46 @@ func setHorizontalScaleAppliedRecommendations(toReplicas float64, ns, targetName
 	)
 }
 
-func unsetHorizontalScaleAppliedRecommendations(ns, targetName, autoscalerName string) {
+func deleteGaugeLinkedToDPA(metric telemetry.Gauge, ns, autoscalerName string) {
 	tags := map[string]string{
-		"namespace":        ns,
-		"target_name":      targetName,
-		"autoscaler_name":  autoscalerName,
-		le.JoinLeaderLabel: le.JoinLeaderValue,
+		"namespace":       ns,
+		"autoscaler_name": autoscalerName,
+	}
+
+	metric.DeletePartialMatch(tags)
+}
+
+// getAutoscalerLabels extracts common labels from a DatadogPodAutoscaler or PodAutoscalerInternal
+func getAutoscalerLabels[T *datadoghq.DatadogPodAutoscaler | model.PodAutoscalerInternal](autoscaler T) []string {
```

Comment:
This is misleading with actual object labels, it should be called `getAutoscalerTelemetryLabels`